DXCDT-141: "Multi-environment Workflows" doc page [6/10] (#572)
auth0/auth0-deploy-cli: docs/multi-environment-workflow.md

> # Incorporating into multi-environment workflows

Reviewer: I'd suggest adding a table of contents

> The Deploy CLI is most powerful when working within a multi-tenant, multi-environment context. When integrated into your CI/CD development workflows, can be used to propagate Auth0 changes from feature development all the way through production.

Reviewer 2: ```suggestion
The Deploy CLI supports working within a multi-tenant, multi-environment context. When integrated into your CI/CD development workflows, can be used to propagate Auth0 changes from feature development all the way through production.
```

IMO the wording implies that it shouldn't be used for anything else than multi env context.

In general, the advised workflow is as follows:

> - Create a separate Auth0 tenant for each environment (ex: dev, staging, prod)

Reviewer: Maybe these could be an ordered list?

- Create a single repository of resource configuration files for all environments
- Add a step in your CI/CD pipeline when deploying to environments that applies the Auth0 resource configurations to the appropriate Auth0 tenant

## Tenant to Environment

It is recommended to have a separate Auth0 tenant/account for each environment you have. For example:

| Environment | Tenant        |
| ----------- | ------------- |
| Development | travel0-dev   |
| Testing     | travel0-uat   |
| Staging     | travel0-stage |
| Production  | travel0-prod  |

## Resource configuration repository

When exported, your Auth0 tenant state will be represented as a set of resource configuration files, either in a [YAML or JSON format](#). In a multi-environment context it is expected to have a single repository of resource configurations that is applied to all environments. In practice, this may exist as a directory in your project’s codebase or in a separate codebase altogether.

> You should have at least one branch for each tenant in your repository, which allows you to make changes without deploying them (the changes would only deploy when you merged your branch into the master, or primary, branch). With this setup, you can have a continuous integration task for each environment that automatically deploys changes to the targeted environment whenever the master branch receives updates.

Reviewer 2: I find this a bit confusing personally. Couldn't I have a branch that updates all the env configs and have both of them deploy when it gets merged to master?

Author: >Couldn't I have a branch that updates all the env configs and have both of them deploy when it gets merged to master?

That would defeat the purpose of the multi-env workflow. By incrementally propagating changes from lower environments all the way to production, you're able to test that your changes work. If you were to change _all_ environment configurations when merging to main, you're deploying to dev, staging and prod all at once.

I think some confusion here might stem from "env config" being named similarly to "resource config".

> Your workflow would, therefore, look something like this:

Reviewer 2: ```suggestion
Your workflow could potentially look something like this:
```

Suggestion
